File: euler3.rb
number = 600851475143
factors = [1]
largest_prime_factor = []
potentially_prime = [0]
not_prime = []

#finding factors
(2..number/2).each do |i|
	if number%i == 0
#		puts "#{i} is a factor of #{number}"
		if factors[factors.length-1] != i &&  factors[factors.length-1] != number/factors.last
			factors.push(i)
			factors.push(number/factors.last)
		else 
			break
		end
	end
end
factors.sort!
factors.reverse
#puts factors



#prime check on factors

(factors.length-1).downto(0) do |l|
	(2..factors[l]/2).each do |k|
		if factors[l+1] == potentially_prime.last
#			puts potentially_prime.last
			largest_prime_factor.push(potentially_prime.last)
#			puts "found a prime!#{potentially_prime.last}"
			break
		end
		if factors[l]%k==0
			not_prime.push(factors[l])
			if potentially_prime.last == factors[l]
				potentially_prime.pop
			end	
#			puts "#{factors[l]} is not a prime"
			break
		elsif potentially_prime.last != factors[l] && not_prime.last != factors[l]
			potentially_prime.push(factors[l])
#			puts "#{factors[l]} is potentially a prime"
		end
	end
end	
	
potentially_prime.sort!
largest_prime_factor = potentially_prime.last
puts "#{largest_prime_factor} is the largest prime factor of #{number}"

File: euler3.js
//function getting prime numbers from input

function getPrimes(num){
  var primes=[2];
  var noPrimes=[1];
  for (var i=3;i<num;i++){
    for (var j=2;j<i;j++){
      if (i%j===0){
        noPrimes.push(i);
        if (primes[primes.length-1] === i){
        primes.pop();
        }else{ }
        break;}
      else {
        if (noPrimes[noPrimes.length-1] !== i && primes[primes.length-1] !== i){
        primes.push(i);} else{}
        }
      }
		}
  return primes;
}
console.log(getPrimes(13195));
// function to get largest prime factor of input (num)

function getLargestPrimeFactor(num){
  var primeFactors = [];
  var largestPrimeFactor = 0;
  var primes = getPrimes(num);

  for (var i=0;i<primes.length;i++) {
    if (num%primes[i] === 0){
      primeFactors.push(primes[i]);
      largestPrimeFactor = primes[i];
    }else{}
  }
  return(largestPrimeFactor)
}
  console.log(getLargestPrimeFactor(13195));
